tests/unit/lms/services/application_instance_service_test.py

```diff
@@ -73,3 +51,25 @@ def application_instances(self):
         # Add some "noise" application instances to the DB for every test, to
         # make the tests more realistic.
         factories.ApplicationInstance.create_batch(size=3)
+
+
+class TestFactory:
+    def test_it(self, ApplicationInstanceService, pyramid_request):
+        application_instance_service = factory(mock.sentinel.context, pyramid_request)
+
+        ApplicationInstanceService.assert_called_once_with(
+            pyramid_request.db, pyramid_request.lti_user.oauth_consumer_key
+        )
+        assert application_instance_service == ApplicationInstanceService.return_value
+
+    def test_it_with_no_lti_user(self, ApplicationInstanceService, pyramid_request):
+        pyramid_request.lti_user = None
+
+        application_instance_service = factory(mock.sentinel.context, pyramid_request)
+
+        ApplicationInstanceService.assert_called_once_with(pyramid_request.db, None)
+        assert application_instance_service == ApplicationInstanceService.return_value
+
+    @pytest.fixture(autouse=True)
+    def ApplicationInstanceService(self, patch):
+        return patch("lms.services.application_instance.ApplicationInstanceService")
```

Comment:
:+1: Cover dem gaps!

A part of me is screaming to parameterise this but... nah, it would end up about as long with a big if statement in the middle